esmvaltool/diag_scripts/ocean/diagnostic_maps_multimodel.py

```diff
@@ -0,0 +1,316 @@
+                diff_cubes.append(cubes[thename][layer] -
+                                  cubes[obsname][layer])
+
+        # data ranges
+        maps_range = diagtools.get_cube_range(maps_cubes)
+        if hasobs:
+            diff_range = diagtools.get_cube_range(diff_cubes)
+            maps_range = diagtools.get_cube_range([cubes[obsname][layer]])
+
+        if 'maps_range' in metadata[filenames[0]]:
+            maps_range = metadata[filenames[0]]['maps_range']
+            extend = 'both'
+        if 'diff_range' in metadata[filenames[0]]:
+            diff_range = metadata[filenames[0]]['diff_range']
+            extend = 'both'
+
+        # create subplots
+
+        varunit = str(maps_cubes[0].units)
+        varname = maps_cubes[0].var_name
+        gsc = gridspec.GridSpec(layout[0], layout[1])
+        yy = 0
+        xx = 0
+        clevels = 13
+        for ii in range(len(maps_cubes)):
+            hascbar = False
+            cube = maps_cubes[ii]
+            thename = name_cubes[ii]
+            nspace = np.linspace(
+                maps_range[0], maps_range[1], clevels, endpoint=True)
+            cmap = 'viridis'
+
+            if thename in [obsname, name_cubes[-1]]:
+                hascbar = True
+
+            if ii == 0:
+                thename = thename + ' (' + varname + ') [' + varunit + ']'
+
+            if (hasobs) & (ii > 0):
+                cube = diff_cubes[ii - 1]
+                nspace = np.linspace(
+                    diff_range[0], diff_range[1], clevels, endpoint=True)
+                cmap = 'RdBu_r'
+                thename = thename
+
+            axs = plt.subplot(gsc[xx, yy], projection=proj)
+            add_map_plot(
+                axs,
+                cube,
+                nspace,
+                yy,
+                cmap=cmap,
+                title=thename,
+                extend=extend,
+                hascbar=hascbar)
+
+            # next row & column indexes
+            xx = xx + 1
+            if xx == layout[0]:
+                xx = 1 if hasobs else 0
+                yy = yy + 1
+
+        # Adjust subplots size & position
+        plt.subplots_adjust(
+            top=0.92,
+            bottom=0.08,
+            left=0.05,
+            right=0.95,
+            hspace=0.15,
+            wspace=0.15)
+
+        # Vertically detach OBS plot and center
+        if hasobs:
+            axs = fig.axes
+            box = axs[0].get_position()
+            shift = box.y0 * 0.05
+            box.y0 = box.y0 + shift
+            box.y1 = box.y1 + shift
+            shift = box.x1 - box.x0
+            box.x0 = 0.5 - shift * 0.5
+            box.x1 = box.x0 + shift
+            axs[0].set_position(box)
+
+        # Determine image filename:
+        fn_list = ['multimodel_vs', obsname, varname, str(layer), 'maps']
+        path = diagtools.folder(cfg['plot_dir']) + '_'.join(fn_list)
+        path = path.replace(' ', '') + image_extention
+
```

Comment:
why would there be empty spaces ' ' in a path?

Reply:
actually this is another leftover ... over safety measure!